client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -12,33 +14,35 @@ interface Params< FlowSteps extends StepperStep[] > {
 }
 
 export const useStepNavigationWithTracking = ( {
-	flow,
+	flow: { useTracksEventProps, useStepNavigation, name: flowName, variantSlug },
 	currentStepRoute,
 	navigate,
 	steps,
 }: Params< ReturnType< Flow[ 'useSteps' ] > > ) => {
-	const stepNavigation = flow.useStepNavigation(
+	const stepNavigation = useStepNavigation(
 		currentStepRoute,
 		navigate,
 		steps.map( ( step ) => step.slug )
```

Comment:
This needs updating. Even if `useStepNavigation` decides to optimize (not within this hook's control), we'd probably break the optimization via a new reference on every call.

Reply:
I think you can go ahead and nuke the `steps` parameter, it's not used anywhere.

Reply:
There is existing usage in `plugin-bundle-flow`. We can certainly refactor that one and nuke the parameter, or work it out differently.

I'll create the issue and address it in a follow-up once we merge. Not touching flows in this PR sounds like a good deal :)

Reply:
Thank you. That flow bums me out :(

Reply:
https://github.com/Automattic/wp-calypso/issues/93808